daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/dfa/TextDelimitedUnparser.scala

```diff
@@ -248,6 +248,8 @@ class TextDelimitedUnparser(override val context: TermRuntimeData)
     escapeEscapeChar: MaybeChar, state: UState): (String, Boolean) = {
     Assert.invariant(delims != null)
     Assert.invariant(field != null)
+    if (hasEscCharAsDelimiter)
+      UnparseError(One(context.schemaFileLocation), One(state.currentLocation), "Delimiters may not begin with the dfdl:escapeCharacter (%s)", escapeChar)
```

Comment:
Surprising this is uncovered. I guess you need a specifically unparserTestCase, because a negative parserTestCase will never get to the unparse.

Reply:
Test has been added that covers this now.  It needed a separator that was determined at runtime during unparsing.